src/shared/util/logic-utils/checkbox/checkbox-response-value-transformer.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable typesafe/no-throw-sync-func */
+import { cloneDeep, omit } from 'lodash'
+
+import {
+  BasicField,
+  FieldSchemaOrResponse,
+  ICheckboxField,
+  ICheckboxResponse,
+  IField,
+  ILogicCheckboxResponse,
+  ILogicClientFieldSchema,
+  ILogicInputClientSchema,
+  IPopulatedForm,
+} from '../../../../types'
+import { isCheckboxField } from '../../../../types/field/utils/guards'
+import { hasProp } from '../../has-prop'
+import { LogicFieldSchemaOrResponse } from '../../logic'
+
+/**
+ * Adaptor function to transform checkbox field response values into checkbox condition shape
+ * @param field Logic field from either the backend or frontend
+ * @param form
+ * @returns Checkbox field response in checkbox condition shape
+ * @throws Error if checkbox value has invalid shape or if field cannot be found in the form
+ */
+export const transformCheckboxForLogic = (
+  field: FieldSchemaOrResponse,
+  formFields: IPopulatedForm['form_fields'],
+): LogicFieldSchemaOrResponse => {
+  field = cloneDeep(field) // clone field to prevent changes to original
+  if (isClientCheckboxValue(field)) {
+    return convertClientCheckboxValue(field, formFields)
+  } else if (isServerCheckboxValue(field)) {
+    return convertServerCheckboxValue(field, formFields)
+  } else {
+    // eslint-disable-next-line typesafe/no-throw-sync-func
+    throw new Error('Checkbox value shape is invalid')
+  }
+}
+
+type ClientCheckboxField = Omit<ILogicInputClientSchema, 'fieldValue'> & {
+  fieldValue: boolean[]
+}
+
+type TransformedClientCheckboxField = ILogicClientFieldSchema
```

Comment:
this type aliasing is pretty confusing, can we just stick to one name for one type?